vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -295,35 +298,18 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     }
     if (!consumes.isEmpty()) {
       // Can this route consume the specified content type
-      String contentType = request.headers().get("content-type");
-      boolean matches = false;
-      for (String ct: consumes) {
-        if (ctMatches(contentType, ct)) {
-          matches = true;
-          break;
-        }
-      }
-      if (!matches) {
-        return false;
-      }
+      MIMEHeader contentType = context.parsedHeaders().contentType();
+      Optional<MIMEHeader> consumal = contentType.findMatchedBy(consumes);
+      return consumal.isPresent();
     }
     if (!produces.isEmpty()) {
-      String accept = request.headers().get("accept");
-      if (accept != null) {
-        List<String> acceptableTypes = Utils.getSortedAcceptableMimeTypes(accept);
-        for (String acceptable: acceptableTypes) {
-          for (String produce : produces) {
-            if (ctMatches(produce, acceptable)) {
-              context.setAcceptableContentType(produce);
-              return true;
-            }
-          }
+      List<MIMEHeader> acceptableTypes = context.parsedHeaders().accept();
```

Comment:
I'm getting a `NullPointerException` here in `RouterTest.testSubRouterNPE` test. The cause is `.parsedHeaders()` returning null.
I looked at `RoutingContextDecorator`, `RoutingContextImpl`, `RoutingContextWrapper` and `RoutingContextBase`.

- `RoutingContextBase` cannot be instantiated directly.
- Both `RoutingContextDecorator` and `RoutingContextWrapper` require a `RoutingContext`
- The only `RoutingContext` that does not require itself is `RoutingContextImpl`.

How come this is null in there? What am I missing when dealing with sub routers?